File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.js
// Last updated: 24/07/2026, 23:55:49
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function (s) {
    let i = 0; j = 0; max = 0
    let set = new Set()
    while (j < s.length) {
        if (set.has(s[j])) {
            max = Math.max(set.size, max)
            while (i < j) {
                if (s[i] == s[j]) {
                    set.delete(s[i])
                    i++
                    break
                }
                set.delete(s[i])
                i++
            }
        } else {
            set.add(s[j])
            j++
        }
    }
    max = Math.max(set.size, max)
    return max
};

File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.java
// Last updated: 24/07/2026, 23:55:54
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashSet <Character> hm = new HashSet <>();
        int i=0,sub=0;

        for(int j=0;j<s.length();j++){
            if(hm.contains(s.charAt(j))){
                while(hm.contains(s.charAt(j))){
                    hm.remove(s.charAt(i));
                    i++;
                }
                hm.add(s.charAt(j));
            }else{
                hm.add(s.charAt(j));
                sub=Math.max(sub,j-i+1);
            }
        }
return sub;
       }}